ArduCopter/Log.cpp

```diff
@@ -86,6 +86,12 @@ void Copter::Log_Write_Control_Tuning()
         terr_alt = DataFlash.quiet_nan();
     }
 #endif
+    float des_alt_m = 0.0f;
```

Comment:
If the entire concept of desired altitude makes no sense it should be logged as NaN, most definitely not as zero.  Zero is in-band, NaN is not, and NaN gives logging and analysis tools more information to deal with.

Reply:
particularly via the `quiet_nanf()` helper to indicate it's the one we expected.